### PR TITLE
Pods 5548:Add Establisher contact details | Company

### PR DESCRIPTION
**Pull request by jigneshkumarpatel**

### CODE REVIEW

File: app/controllers/establishers/company/contact/WhatYouWillNeedCompanyContactController.scala [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    with Retrievals
+    with NunjucksSupport {
+
+  def onPageLoad(index: Index): Action[AnyContent] =
+    (authenticate andThen getData andThen requireData).async {
+      implicit request =>
+        CompanyDetailsId(index).retrieve.right.map {
+          details =>
+            renderer.render(
+              template = "establishers/company/contact/whatYouWillNeed.njk",
```
Comment (madhuranakate) [inline app/controllers/establishers/company/contact/WhatYouWillNeedCompanyContactController.scala:55]: can this be put in a common place to be reused for trustee company contact?

File: app/identifiers/establishers/company/contact/EnterEmailId.scala [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import play.api.libs.json.{Format, JsPath, Json}
+
+case class EnterEmailId(index: Int) extends TypedIdentifier[String] {
+  override def path: JsPath =
+    EstablishersId(index).path \ EnterEmailId.toString
+}
+
+object EnterEmailId {
+  override lazy val toString: String =
+    "companyemail"
```
Comment (madhuranakate) [inline app/identifiers/establishers/company/contact/EnterEmailId.scala:30]: either use just "email" or change to camelCase

File: app/identifiers/establishers/company/contact/EnterPhoneId.scala [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+import play.api.libs.json.{Format, JsPath, Json}
+
+case class EnterPhoneId(index: Int) extends TypedIdentifier[String] {
+  override def path: JsPath =
+    EstablishersId(index).path \ EnterPhoneId.toString
+}
+
+object EnterPhoneId {
+  override lazy val toString: String =
+    "companyphone"
```
Comment (madhuranakate) [inline app/identifiers/establishers/company/contact/EnterPhoneId.scala:30]: either use just "phone" or change to camelCase

File: test/utils/datacompletion/DataCompletionEstablishersSpec.scala [hunk 4]
```diff
@@ -173,5 +174,28 @@ class DataCompletionEstablishersSpec
...
+
+        ua.isEstablisherCompanyContactDetailsCompleted(0).value mustBe false
+      }
+
+      "return None when no answer is present" in {
+        UserAnswers().isEstablisherCompanyContactDetailsCompleted(0) mustBe None
+      }
+    }
+
   }
```
Comment (madhuranakate) [inline test/utils/datacompletion/DataCompletionEstablishersSpec.scala:200]: SpokeCreationServiceSpec needs new tests/updates for updated code

**Review by madhuranakate — CHANGES_REQUESTED** [review]

File: app/controllers/establishers/company/contact/WhatYouWillNeedCompanyContactController.scala (hunk 1 above)
Comment (jigneshkumarpatel) [reply app/controllers/establishers/company/contact/WhatYouWillNeedCompanyContactController.scala]: Move the njk to establishers and added title on controller to have component use across module .

File: app/identifiers/establishers/company/contact/EnterEmailId.scala (hunk 2 above)
Comment (jigneshkumarpatel) [reply app/identifiers/establishers/company/contact/EnterEmailId.scala]: updated to compnayEmail

File: app/identifiers/establishers/company/contact/EnterPhoneId.scala (hunk 3 above)
Comment (jigneshkumarpatel) [reply app/identifiers/establishers/company/contact/EnterPhoneId.scala]: updated to camelCase

File: test/utils/datacompletion/DataCompletionEstablishersSpec.scala (hunk 4 above)
Comment (jigneshkumarpatel) [reply test/utils/datacompletion/DataCompletionEstablishersSpec.scala:200]: updated test class for Spokeservice

**Review by madhuranakate — APPROVED** [review]

**Pull request merged**